### DISCUSSION TITLE
Progress Sync between ABS and ThirdPartyClients like Shelfplayer

### DISCUSSION THREAD

**Discussion by alexanderkuenzel** [advplyr/audiobookshelf#3538] [Q&A]
Why does instant sync of playback progress work if audiobooks are streamed, but if they are downloaded, the progress is only synced after reopening the app. And ABS recognizes the change only if the audiobook is started directly on server and not in realtime. Is there a technical reason for this or is it just a bug?

**Comment by sevenlayercookie** [comment]
> Why does instant sync of playback progress work if audiobooks are streamed, but if they are downloaded, the progress is only synced after reopening the app. And ABS recognizes the change only if the audiobook is started directly on server and not in realtime. Is there a technical reason for this or is it just a bug?

My understanding is this is how ShelfPlayer is programmed. There is an API call available that syncs the offline session with the server; it's up to the developer of the app (ShelfPlayer) to call that. For my Alexa skill for example, I call the update session API every time pause or play is pressed, or when the skill is closed.
